integration-tests/src/tests/client/features/increase_storage_compute_cost.rs

```diff
@@ -0,0 +1,369 @@
+
+/// Test that `storage_write` compute limit is respected in new version.
+#[test]
+fn test_storage_write() {
+    //  `insert_strings(from: u64, to: u64)` makes (`to` - `from`) `storage_write` calls.
+    let method_name = "insert_strings".to_owned();
+    let num_writes = 100u64;
+    let method_args: Vec<u8> =
+        0u64.to_le_bytes().into_iter().chain(num_writes.to_le_bytes()).collect();
+    let num_transactions = 200;
+    let uses_storage = true;
+    let fails = false;
+    assert_compute_limit_reached(method_name, method_args, num_transactions, uses_storage, fails);
+}
+
+/// Test that `storage_remove` compute limit is respected in new version.
+#[test]
+fn test_storage_remove() {
+    //  `delete_strings(from: u64, to: u64)` makes (`to` - `from`) `storage_remove` calls.
+    let method_name = "delete_strings".to_owned();
+    let num_deletes = 1000u64;
+    let method_args: Vec<u8> =
+        0u64.to_le_bytes().into_iter().chain(num_deletes.to_le_bytes()).collect();
+    let num_transactions = 100;
+    let uses_storage = true;
+    let fails = false;
+    assert_compute_limit_reached(method_name, method_args, num_transactions, uses_storage, fails);
+}
+
+/// Test that `storage_write` compute limit is respected in new version,
+/// specifically when running out of gas.
+#[test]
+fn test_storage_write_gas_exceeded() {
+    //  `insert_strings(from: u64, to: u64)` makes (`to` - `from`) `storage_write` calls.
+    let method_name = "insert_strings".to_owned();
+    // 10000 writes should be too much and result in gas exceeded
+    let num_writes = 10000u64;
+    let method_args: Vec<u8> =
+        0u64.to_le_bytes().into_iter().chain(num_writes.to_le_bytes()).collect();
+    let num_transactions = 10;
+    let uses_storage = true;
+    let fails = true;
+    assert_compute_limit_reached(method_name, method_args, num_transactions, uses_storage, fails);
+}
+
+/// Check receipts that don't touch storage are unaffected by the new compute costs.
+#[test]
+fn test_non_storage() {
+    // `sum_n(u64)` just does some WASM computation.
+    // It should not be affected by compute costs, as it doesn't access storage.
+    let method_name = "sum_n".to_owned();
+    // note: If we make `method_args` only 1M, then the gas cost increase due to
+    // finite-wasm change (also included in version 61) is large enough that we
+    // see the number of receipts fitting in a chunk going from 64 to 61 even
+    // without the compute costs. 10.03M is the sweet spot where the granularity is
+    // course enough that the finite-wasm change doesn't interfere. We can
+    // process exactly 8 receipts per chunk before and after finite-wasm.
+    // And if compute costs would additionally affect this, the test would fail
+    // because it is very close at the border between 7 or 8 receipts fitting
+    // (7 receipts with finite wasm require 999 Tgas, 8 receipts require 1142 Tgas).
+    let method_args: Vec<u8> = 10_030_000u64.to_le_bytes().to_vec();
```

Comment:
Maybe we could tweak gas limit in Genesis to simplify computations, but I don't mind if this setup works already.

Reply:
Yeah, that's a nice idea. But I think I will merge this as-is to not delay further.